homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+template<typename T>
+class unique_ptr {
+public:
+    unique_ptr() = default;
+    unique_ptr(T* pointer) : pointer_(pointer) {}
+    unique_ptr(unique_ptr& ptr);
+    unique_ptr(unique_ptr&& ptr);
+    ~unique_ptr();
+    T& operator*() const { return *pointer_; }
+    T* operator->() const { return pointer_; }
+    T* get() const { return pointer_; }
+    T* release();
+    void reset(T* new_ptr);
```

Comment:
```suggestion
    void reset(T* new_ptr = nullptr);
```
The default value will be useful here